parser/src/powdr.lalrpop

```diff
@@ -202,24 +202,17 @@ pub InstructionBody: InstructionBody<T> = {
     "=" <Identifier> "." <Identifier> => InstructionBody::External(<>),
 }
 
-InstructionBodyElements: Vec<InstructionBodyElement<T>> = {
+InstructionBodyElements: Vec<PilStatement<T>> = {
     <mut list:( <InstructionBodyElement> "," )*> <end:InstructionBodyElement>  => { list.push(end); list },
     => vec![]
 }
 
-InstructionBodyElement: InstructionBodyElement<T> = {
-    <l:Expression> "=" <r:Expression> => InstructionBodyElement::PolynomialIdentity(l, r),
-    <SelectedExpressions> <PlookupOperator> <SelectedExpressions> => InstructionBodyElement::PlookupIdentity(<>),
-    <id:Identifier> "(" <arguments:ExpressionList> ")" => InstructionBodyElement::FunctionCall(FunctionCall {<>})
-}
-
-// This is only valid in instructions, not in PIL in general.
-// "connect" is not supported because it does not support selectors
-// and we need that for the instruction.
-
-PlookupOperator: PlookupOperator = {
-    "in" => PlookupOperator::In,
-    "is" => PlookupOperator::Is,
+InstructionBodyElement: PilStatement<T> = {
+    PolynomialIdentity,
```

Comment:
I didn't allow all statements. I think this is fine for now.

Reply:
It may be better to allow everything here and fail in the type checker. It simplifies the parser and makes the errors cleaner.

Reply:
I'm afraid I will run into similar errors like the one for FunctionCallStatement. The Type checker already contains the checks.